apps/core/scenes/search/Search.js

```diff
@@ -288,56 +201,34 @@ class Search extends React.Component<Props, State> {
               />
             </View>
             <View>
-              <TripInput
-                onPress={this.handlePlacePress}
-                label="From"
-                icon={<Icon name="airplane-takeoff" />}
-                value={travelFrom}
-              />
-              <TouchableWithoutFeedback onPress={this.handlePlaceSwitchPress}>
-                <View style={styles.placeSwitch}>
-                  <Icon name="replace" color="#7F91A8" />
-                </View>
-              </TouchableWithoutFeedback>
-              <TripInput
-                onPress={this.handlePlacePress}
-                label="To"
-                icon={<Icon name="airplane-landing" />}
-                value={travelTo}
+              <Placepickers
+                travelFrom={travelFrom}
+                travelTo={travelTo}
+                handlePlacePress={this.handlePlacePress}
+                handlePlaceSwitchPress={this.handlePlaceSwitchPress}
               />
             </View>
-            <TripInput
-              onPress={this.handleDepartureDatePress}
-              label="Departure"
-              icon={<Icon name="calendar" />}
-              value={getFormatedDate(dateFrom)}
+            <Datepickers
```

Comment:
Modal component is in my opinion broken. 

I wanted to create a TripTypeComponent containing TripTypeButton and modal, and one component PassengerPicker containing PassengerButton and modal. 

Then I noticed on mobile that shadow was wrong. 

This component needs to be improved. Now it is quite useless, since you need to have it on root level. 

I will try to make it look good for web as well...

Reply:
Modal can be definitely improved. We can probably remove black overlay `View` which is causing current limitation that you've mentioned. The only purpose of it was to prevent ugly slide-in overlay effect which is currently visible on mobile `DatePicker`.
![slide](https://user-images.githubusercontent.com/2660330/51752929-e506a500-20b8-11e9-8a1a-52f89b5cb900.gif)
But we can probably solve it by fading black overlay in after slide-in animation or something similar.

Reply:
Well, we have this solved in `mobile` so 🤷‍♂️